platform/core/src/classes/MetadataProvider.js

```diff
@@ -7,6 +7,10 @@ import fetchPaletteColorLookupTableData from '../utils/metadataProvider/fetchPal
 import toNumber from '../utils/toNumber';
 import combineFrameInstance from '../utils/combineFrameInstance';
 
+import dicomImageLoader from '@cornerstonejs/dicom-image-loader';
+import dcmjs from 'dcmjs';
```

Comment:
I thought these are new dependencies but seems like we already have them although I prefer if we can find a way to not call dicomImageLoader in the provider here, any thoughts?

Reply:
What I would suggest is having a lazy load function to call that if it is defined in the metadata, it can be called to replace the metadata with a full load version of it - which then removes the function from the metadata.  That way, different providers can provide different versions of the function, some of which might do the dcmjs thing, and others might load JSON metadata, while others might just read bulkdata information which is missing.

Reply:
I like this